wazo_phoned/plugin_helpers/client/http.py

```diff
@@ -0,0 +1,247 @@
+            term=term,
+            limit=limit,
+            total=total_results,
+            offset=offset,
+            offset_next=self._next_offset(offset, limit, total_results),
+            offset_previous=self._previous_offset(offset, limit),
+        )
+
+        return Response(response_rendered, content_type=self.content_type, status=200)
+
+    def _next_offset(self, offset, limit, results_count):
+        if limit is None:
+            return None
+
+        next_offset = offset + limit
+        if next_offset >= results_count:
+            return None
+
+        return next_offset
+
+    def _previous_offset(self, offset, limit):
+        if offset == 0:
+            return None
+
+        if limit is None:
+            return None
+
+        previous_offset = offset - limit
+        if previous_offset < 0:
+            return 0
+
+        return previous_offset
+
+    def _get_user_tenant_uuid(self, user_uuid):
+        try:
+            return self.auth_client.users.get(user_uuid)['tenant_uuid']
+        except HTTPError as e:
+            response = getattr(e, 'response', None)
+            status_code = getattr(response, 'status_code', None)
+            if status_code == 404:
+                raise NoSuchUser(user_uuid)
+            raise
+        except ConnectionError:
+            raise WazoAuthConnectionError()
+
+
+def _build_next_url(current):
+    if current == 'menu':
+        return request.base_url.replace('menu', 'input', 1)
+    if current == 'input':
+        return request.base_url.replace('input', 'lookup', 1)
+    if current == 'lookup':
+        return request.base_url
+    return None
+
+
+class _PhoneResultFormatter:
+
+    _NAME_FIELD = 'name'
+    _NUMBER_FIELD = 'number'
+    _INVALID_CHARACTERS_REGEX = re.compile(r'[^\d*#+\(\)]+')
+    _SPECIAL_NUMBER_REGEX = re.compile(r'^\+(\d+)\(\d+\)(\d+)$')
+    _PARENTHESES_REGEX = re.compile(r'[\(\)]')
+
+    def __init__(self, lookup_results):
+        self._lookup_results = lookup_results
+        self._number_fields = self._extract_number_fields()
+        self._name_field_position = self._find_name_field_position()
+
+    def format_results(self):
+        results = []
+        for lookup_result in self._lookup_results['results']:
+            self._format_result(lookup_result, results)
+        return results
+
+    def _format_result(self, result, out):
+        for display_name, number in self._extract_result(result):
+            number = self._normalize_number(number)
+            if not number:
+                continue
+            out.append(_PhoneFormattedResult(display_name, number))
+
+    def _normalize_number(self, number):
+        number = self._extract_number_from_pretty_number(number)
+        return number
+
+    def _extract_result(self, result):
+        name = result['column_values'][self._name_field_position]
+        first_number = True
+        for candidate in self._number_fields:
+            if first_number:
```

Comment:
Why the first number do not have `(..)` ?
If there are more than one number, then all number should have `(...)`

Reply:
It was to emulate the existing behaviour of dird phone plugins. The mobile number column has a `number_display` with default value `{} (mobile)` on dird, which is a format string. See https://github.com/wazo-platform/wazo-dird/blob/master/wazo_dird/plugins/phone_helpers.py#L117
The first number does not have a specific `number_display` by default. It would indeed be more consistent to put the column name between parentheses for every number when there are more than two to display.